### ISSUE TITLE
Error on query builder when using client_credentials to access routes with auth:api protection 

### ISSUE THREAD

**Issue by danteldev** [laravel/passport#311]
When I try to access a route with the auth:api middleware using a token created with client_credentials grant type, instead of getting a response `{"error":"Unauthenticated."}` I get this exception.
So when TokenGuard try to get a user through the EloquentUserProvider sending a empty string (because there is no user on client_credentials token), postgres throws an exception because it doesn't cast '' to int.

> QueryException in Connection.php line 647:
> SQLSTATE[22P02]: Invalid text representation: 7 ERROR: invalid input syntax for integer: "" (SQL: select * from "users" where "users"."id" = limit 1)
> in Connection.php line 647
> at Connection->runQueryCallback('select * from "users" where "users"."id" = ? limit 1', array(''), object(Closure)) in Connection.php line 607
> at Connection->run('select * from "users" where "users"."id" = ? limit 1', array(''), object(Closure)) in Connection.php line 326
> at Connection->select('select * from "users" where "users"."id" = ? limit 1', array(''), true) in Builder.php line 1718
> at Builder->runSelect() in Builder.php line 1703
> at Builder->get(array('*')) in Builder.php line 493
> at Builder->getModels(array('*')) in Builder.php line 477
> at Builder->get(array('*')) in Builder.php line 409
> at Builder->first(array('*')) in Builder.php line 284
> at Builder->find('') in EloquentUserProvider.php line 47
> at EloquentUserProvider->retrieveById('') in TokenGuard.php line 116
> at TokenGuard->authenticateViaBearerToken(object(Request)) in TokenGuard.php line 90
> at TokenGuard->user(object(Request)) in PassportServiceProvider.php line 251
> at PassportServiceProvider->Laravel\Passport\{closure}(object(Request))
> at call_user_func(object(Closure), object(Request)) in RequestGuard.php line 54
> at RequestGuard->user() in GuardHelpers.php line 49
> at RequestGuard->check() in Authenticate.php line 61
> at Authenticate->authenticate(array('api')) in Authenticate.php line 41
> at Authenticate->handle(object(Request), object(Closure), 'api') in Pipeline.php line 148
> at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
> at Pipeline->Illuminate\Routing\{closure}(object(Request)) in ThrottleRequests.php line 49
> at ThrottleRequests->handle(object(Request), object(Closure), '60', '1') in Pipeline.php line 148
> at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
> at Pipeline->Illuminate\Routing\{closure}(object(Request)) in Pipeline.php line 102
> at Pipeline->then(object(Closure)) in Router.php line 561
> at Router->runRouteWithinStack(object(Route), object(Request)) in Router.php line 520
> at Router->dispatchToRoute(object(Request)) in Router.php line 498
> at Router->dispatch(object(Request)) in Kernel.php line 174
> at Kernel->Illuminate\Foundation\Http\{closure}(object(Request)) in Pipeline.php line 30
> at Pipeline->Illuminate\Routing\{closure}(object(Request)) in TransformsRequest.php line 30
> at TransformsRequest->handle(object(Request), object(Closure)) in Pipeline.php line 148
> at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
> at Pipeline->Illuminate\Routing\{closure}(object(Request)) in TransformsRequest.php line 30
> at TransformsRequest->handle(object(Request), object(Closure)) in Pipeline.php line 148
> at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
> at Pipeline->Illuminate\Routing\{closure}(object(Request)) in ValidatePostSize.php line 27
> at ValidatePostSize->handle(object(Request), object(Closure)) in Pipeline.php line 148
> at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
> at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckForMaintenanceMode.php line 46
> at CheckForMaintenanceMode->handle(object(Request), object(Closure)) in Pipeline.php line 148
> at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in Pipeline.php line 53
> at Pipeline->Illuminate\Routing\{closure}(object(Request)) in Pipeline.php line 102
> at Pipeline->then(object(Closure)) in Kernel.php line 149
> at Kernel->sendRequestThroughRouter(object(Request)) in Kernel.php line 116
> at Kernel->handle(object(Request)) in index.php line 53
> at require_once('/home/dantel/Projects/api-cotacoesecompras/public/index.php') in server.php line 21

EloquentUserProvider implements the retrieveById this way:
```php
public function retrieveById($identifier)
    {
        return $this->createModel()->newQuery()->find($identifier);
    }
```

That is basically: 
```php
$model = new \App\User(); 
$model->newQuery()->find('');
```

Which works fine in sqlite, but throws an exception in postgres.
Is there any way of solving this without changing code in Passport?

**Comment by JuaumCruz** [comment]
I believe the problem is in the framework itself that does not check (or convert) the data type before doing the comparison. It has access to the model and could use the incrementing property to check if the value should be converted to int before the comparison.

I made this modification, tested it and started to work normally.
Here is the pull request link I made with the suggested modification: https://github.com/laravel/framework/pull/18364
